extensions/stats/plugin.cc

```diff
@@ -576,9 +576,7 @@ bool PluginRootContext::report(::Wasm::Common::RequestInfo& request_info,
     // been no error in connection.
     uint64_t response_flags = 0;
     getValue({"response", "flags"}, &response_flags);
-    if (peer_node == nullptr &&
-        peer_id != ::Wasm::Common::kMetadataNotFoundValue &&
```

Comment:
@gargnupur I think this was added by you: can you take a look?

Reply:
Speaking it out loud, there are 4 scenarios here:
1) Metadata Exchange has not happened yet -> so return false, so that you come back to report again -> peer_found == false && response_flags => if condition works
2) There is an error(response_flags!=0),  so don't worry about MX and just log -> response_flags != 0 => if condition works as it will go ahead and log
3) Metadata Exchange was attempted but peer not found -> peer_id = kMetadataNotFoundValue -> peer_found = true => if condition works
4) MX has happened, peer is there, no error and thus just log -> peer_found = true -> if condition works

Thus, this will work...